PlayCover/Utils/KeyCover.swift

```diff
@@ -0,0 +1,313 @@
+        }
+        // zip up the key folder
+        // make sure to only compress the key folder, not the entire path to it
+        let source = keyFolderPath.appendingPathComponent(appBundleID)
+        let destination = keyFolderPath.appendingPathComponent("\(appBundleID).zip")
+        let task = Process()
+        task.launchPath = "/usr/bin/zip"
+        task.currentDirectoryPath = keyFolderPath.path
+        task.arguments = ["-q", "-r", destination.path, source.lastPathComponent]
+        task.launch()
+        task.waitUntilExit()
+
+        // encrypt the zip file
+        let task2 = Process()
+        task2.launchPath = "/usr/bin/openssl"
+        task2.currentDirectoryPath = keyFolderPath.path
+        task2.arguments = ["enc", "-aes-256-cbc", "-A",
+                            "-in", destination.path,
+                            "-out", encryptedKeyFile.path,
+                            "-k", KeyCover.shared.keyCoverPlainTextKey!]
+        task2.launch()
+        task2.waitUntilExit()
+
+        // delete the zip file
+        try? FileManager.default.removeItem(at: destination)
+
+        // delete the key folder
+        try? deleteKeyFolder()
+
+        Task { @MainActor in
+            KeyCoverObservable.shared.update()
+        }
+    }
+
+    func decryptKeyFolder() throws {
+        if KeyCover.shared.keyCoverPlainTextKey == nil {
+            return
+        }
+        // decrypt the zip file
+        let task = Process()
+        task.launchPath = "/usr/bin/openssl"
+        task.arguments = ["enc", "-aes-256-cbc", "-A", "-d", "-in", encryptedKeyFile.path, "-out",
+                          keyFolderPath.appendingPathComponent("\(appBundleID).zip").path,
+                          "-k", KeyCover.shared.keyCoverPlainTextKey!]
+        task.launch()
+        task.waitUntilExit()
+
+        // unzip the zip file
+        let task2 = Process()
+        task2.launchPath = "/usr/bin/unzip"
+        task2.currentDirectoryPath = keyFolderPath.path
+        task2.arguments = ["-qq", "-o", keyFolderPath.appendingPathComponent("\(appBundleID).zip").path,
+                           "-d", keyFolderPath.path]
+        task2.launch()
+        task2.waitUntilExit()
+
+        // delete the zip file
+        try? FileManager.default.removeItem(at: keyFolderPath.appendingPathComponent("\(appBundleID).zip"))
+
+        // delete the encrypted key file
+        try? FileManager.default.removeItem(at: encryptedKeyFile)
+
+        Task { @MainActor in
+            KeyCoverObservable.shared.update()
+        }
+    }
+
+    func deleteKeyFolder() throws {
+        try? FileManager.default.removeItem(at: keyFolderPath.appendingPathComponent(appBundleID))
+    }
+
+    func wipeEncryptedKeyFile() throws {
+        try? FileManager.default.removeItem(at: encryptedKeyFile)
+    }
+}
+
+class KeyCoverMaster {
+    static let shared = KeyCoverMaster()
+
+    func setMasterKey(_ key: String) {
+        let tag = "com.playcover.masterkey"
```

Comment:
Should this not be `io.playcover.masterkey` since PlayCover's domain is `.io`?

Reply:
Oops. I forgot